validator.go

```diff
@@ -233,12 +233,11 @@ func (mv *Validator) Validate(x interface{}) error {
 	// TODO calculate likely size of path and pathStack; or alternatively
 	// we could keep validateState instances around in a sync.Pool
 	// to avoid the allocations.
-	state := &validateState{
-		path:      make([]byte, 0, 20),
-		pathStack: make([]int, 0, 10),
-	}
+	state := getState()
 	validate(sv, state)
-	return state.finalError()
+	err := state.finalError()
```

Comment:
Is there a reason why this happens before `putState`?

Reply:
@aranw I'm guessing it's because `putState` clears the errors to nil, so you'd lose them if you ordered it differently.